h2o-admissibleml/src/main/java/hex/Infogram/InfogramUtils.java

```diff
@@ -0,0 +1,239 @@
+    for (String oneCol : colNames)
+      if (train.vec(oneCol).isBad() || train.vec(oneCol).isConst()) 
+        excludeCols.add(oneCol);
+      
+    colNames.removeAll(excludeCols); // remove constant and bad columns
+    return colNames.toArray(new String[colNames.size()]);
+  }
+
+  /**
+   * Method to run infogram model once in order to get the variable importance of the topK predictors
+   * @param parms
+   * @param trainFrame
+   * @param eligiblePredictors
+   * @return
+   */
+  public static String[] extractTopKPredictors(InfogramParameters parms, Frame trainFrame,
+                                               String[] eligiblePredictors, List<Key<Frame>> generatedFrameKeys) {
+    if (parms._top_n_features >= eligiblePredictors.length) return eligiblePredictors;
+    Frame topTrain = extractTrainingFrame(parms, eligiblePredictors, 1, trainFrame);
+    generatedFrameKeys.add(topTrain._key);
+    parms._infogram_algorithm_parameters._train = topTrain._key;
+    ModelBuilder builder = ModelBuilder.make(parms._infogram_algorithm_parameters);
+    Model builtModel = (Model) builder.trainModel().get();
+    Scope.track_generic(builtModel);
+    TwoDimTable varImp = extractVarImp(parms._algorithm, builtModel);
+    String[] ntopPredictors = new String[parms._top_n_features];
+    String[] rowHeaders = varImp.getRowHeaders();
+    System.arraycopy(rowHeaders, 0, ntopPredictors, 0, parms._top_n_features);
+    return ntopPredictors;
+  }
+
+  public static TwoDimTable extractVarImp(InfogramParameters.Algorithm algo, Model model) {
+    switch (algo) {
+      case gbm : return ((GBMModel) model)._output._variable_importances;
+      case glm : return ((GLMModel) model)._output._variable_importances;
+      case deeplearning : return ((DeepLearningModel) model)._output._variable_importances;
+      case drf : return ((DRFModel) model)._output._variable_importances;
+      case xgboost : return ((XGBoostModel) model)._output._variable_importances;
+      default : return null;
+    }
+  }
+
+  /**
+   * This method will perform two functions:
+   * - if user only wants a fraction of the training dataset to be used for infogram calculation, we will split the
+   *   training frame and only use a fraction of it for infogram training purposes;
+   * - next, a new training dataset will be generated containing only the predictors in predictors2Use array. 
+   *
+   * @param parms
+   * @param sensitivePredictors
+   * @param dataFraction
+   * @param
+   * @return
+   */
+  public static Frame extractTrainingFrame(InfogramParameters parms, String[] sensitivePredictors, double dataFraction,
+                                           Frame trainFrame) {
+    if (dataFraction < 1) {  // only use a fraction training data for speedup
+      SplitFrame sf = new SplitFrame(trainFrame, new double[]{parms._data_fraction, 1-parms._data_fraction},
+              new Key[]{Key.make("train.hex"), Key.make("discard.hex")});
```

Comment:
doesn't look safe to use a hardcoded key here, I'd rather suggest to prefix or suffix the passed key:
```java
new Key[]{Key.make("ig_train_"+trainFrame._key), Key.make("ig_discard_"+trainFrame._key)});
```

Reply:
+1, or just use `null` if you don't care about the keys

Reply:
Good idea.  Done.